package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -0,0 +1,515 @@
+        continuous = util.group_consecutive_integers(helices.resindices)
+
+        # must have at least 9 continuous residues for helanal
+        selections = []
+        universe = secondary_structure._universe
+        all_res = universe.residues
+        for resindices in continuous:
+            if len(hel) >= 9:
+                res = all_res[resindices]
+                sel = 'resid {}:{} and {}'.format(res[0].resid,
+                                                  res[-1].resid,
+                                                  select)
+                selections.append(sel)
+        if not selections:
+            raise ValueError('Could not find any helices with at least '
+                             '9 continuous residues. Consider running '
+                             'HELANAL manually')
+
+        return cls(universe, select=selections, ref_axis=ref_axis,
+                   verbose=verbose, flatten_single_helix=flatten_single_helix)
+
+    def __init__(self, universe, select='name CA', ref_axis=[0, 0, 1],
+                 verbose=False, flatten_single_helix=True):
+        super(HELANAL, self).__init__(universe.universe.trajectory,
+                                      verbose=verbose)
+        selections = util.asiterable(select)
+        self.atomgroups = [universe.select_atoms(s) for s in selections]
+        for s, ag in zip(selections, self.atomgroups):
+            ids, counts = np.unique(ag.resindices, return_counts=True)
+            if np.any(counts > 1):
+                dup = ', '.join(map(str, ids[counts > 1]))
+                warnings.warn('Your selection {} includes multiple atoms '
+                              'for residues with these resindices: {}.'
+                              'HELANAL is designed to work on one carbon-alpha '
+                              'per residue.'.format(s, dup))
+        self.ref_axis = np.asarray(ref_axis)
+        self._flatten = flatten_single_helix
+
+    def _zeros_per_frame(self, *dims, n_positions=0):
+        """Create zero arrays where first 2 dims are n_frames, n_values"""
+        first = dims[0]
+        return np.zeros((self.n_frames, first+n_positions, *dims[1:]),
+                         dtype=np.float64)
+
+    def _prepare(self):
+        n_res = [len(ag) for ag in self.atomgroups]
+
+        for key, dims in self.attr_shapes.items():
+            empty = [self._zeros_per_frame(
+                *dims, n_positions=n) for n in n_res]
+            setattr(self, key, empty)
+
+        self.global_axes = [self._zeros_per_frame(3) for n in n_res]
+        self.all_bends = [self._zeros_per_frame(n-3, n-3) for n in n_res]
+
+    def _single_frame(self):
+        _f = self._frame_index
+        for i, ag in enumerate(self.atomgroups):
+            results = helix_analysis(ag.positions)
+            for key, value in results.items():
+                attr = getattr(self, key)
+                attr[i][_f] = value
```

Comment:
haha this melted my brain for a minute but it's very clever

Reply:
Thanks @richardjgowers -- this is because I'm lazy and kept changing what the properties were called to try to make them clearer. However if it impairs readability I'm happy to make it a longform

```
self.local_twists[i][self._frame_index] = results['local_twists']
...
```